pandas/core/indexes/base.py

```diff
@@ -1205,7 +1206,7 @@ def _maybe_disallow_fill(self, allow_fill: bool, fill_value, indices) -> bool:
         """
 
     @Appender(_index_shared_docs["repeat"] % _index_doc_kwargs)
-    def repeat(self, repeats, axis=None):
+    def repeat(self, repeats, axis: AxisInt | None = None):
```

Comment:
I would prefer to change it to `axis: None = None` as the [docs](https://pandas.pydata.org/docs/reference/api/pandas.Index.repeat.html) say:

> Must be None. Has no effect but is accepted for compatibility with numpy.

Reply:
Same applies to the other two changes

Reply:
The doc-string in IndexOpsMixin.argmin/max also says that it is only there for compatibility. So I would be tempted to change even  IndexOpsMixin.argmin/max to `axis: None = None`. One could also make the argument that the type should be `int | None` as both are valid :) Let's make the change for at least repeat to be in sync with the doc-string but the other cases are up to you.

@mroeschke `IndexOpsMixin.argmin` "uses" axis exactly once - `nv.validate_minmax_axis(axis)`. Testing an unused argument seems not very helpful but removing it probably changes the runtime behavior when people expect to get an error.

Reply:
Yeah ideally this should still error the same if a non `None` axis was passed